WordPress/gutenberg
# Components: Make the Popover.Slot optional

Related #53874

## What and why?

Gutenberg can be used as a platform/framework to build block editors. Mainly thanks to the @wordpress/block-editor package. That said, the experience today is not as straightforward as it can be. There can be a lot of small gotchas and hacks you need to do in order to achieve the desired result. One of these small things is the need to manually render `Popover.Slot` somewhere in the React tree.

I think ideally, Popover components should work properly out of the box without any provider or slot, and only if we really want to control their position in the DOM, we should use `Popover.Slot`.

This PR does that by using an "automatic container" for all popovers that is appended to the end of the "body" of the document if there's no slot that is defined manually using `Popover.Slot`. I think that's a better default than inlining popovers which breaks all the styles and more.

**Todo:**

 - [x] I've noticed that when the anchor is within an iframe, the popover is not offset properly to the position of the iframe (you can test for instance the appender of the navigation block). I'm not entirely sure how to fix that because I thought we had that offsetting in place already. cc @talldan @ciampo 

## Testing Instructions

1- Check that the stories updates (removal of popover and slot provider) still work properly.
2- Navigate the editor and ensure popovers, tooltips, drodowns are shown properly.
